app/src/org/commcare/fragments/personalId/PersonalIdBiometricConfigFragment.java

```diff
@@ -265,13 +266,18 @@ private NavDirections navigateToBiometricEnrollmentFailed() {
     }
 
     private void navigateToMessageDisplayForSecurityConfigurationFailure(String errorMessage) {
-        NavDirections navDirections =
-                PersonalIdBiometricConfigFragmentDirections.actionPersonalidBiometricConfigToPersonalidMessage(
-                        getString(R.string.configuration_process_failed_title),
-                        errorMessage,
-                        ConnectConstants.PERSONALID_DEVICE_CONFIGURATION_FAILED, getString(R.string.ok),
-                        null).setIsCancellable(false);
-        Navigation.findNavController(requireView()).navigate(navDirections);
+        Logger.log("NavCheck", "Current destination: " +
+                Navigation.findNavController(requireView()).getCurrentDestination().getLabel());
+        NavController navController = Navigation.findNavController(requireView());
+        if (navController.getCurrentDestination().getId() != R.id.personalid_message_display) {
```

Comment:
This can't be a fix here, I prefer crashing to user clicking on a button that doesn't take them to the next page.